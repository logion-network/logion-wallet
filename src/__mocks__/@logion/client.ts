import { CheckResultType } from "@logion/client/dist/CollectionItem.js";
import { DefaultSignAndSendStrategy } from "@logion/client/dist/Signer.js";

import {
    LogionClient,
    AccountTokens,
    toIsoString,
    fromIsoString,
    ClosedCollectionLoc,
    DraftRequest,
    EditableRequest,
    OpenLoc,
    LocRequestState,
    LocsState,
    HashOrContent,
} from '../LogionClientMock';

export {
    LogionClient,
    AccountTokens,
    toIsoString,
    fromIsoString,
    CheckResultType,
    ClosedCollectionLoc,
    DraftRequest,
    EditableRequest,
    OpenLoc,
    DefaultSignAndSendStrategy,
    LocRequestState,
    LocsState,
    HashOrContent,
}

export { MimeType } from "@logion/client/dist/Mime.js";
