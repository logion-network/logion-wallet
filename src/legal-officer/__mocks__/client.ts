import {
    closeLoc,
    deleteLink,
    voidLoc,
    getVerifiedThirdPartySelections,
} from "./ClientMock";

export {
    closeLoc,
    deleteLink,
    voidLoc,
    getVerifiedThirdPartySelections,
}
