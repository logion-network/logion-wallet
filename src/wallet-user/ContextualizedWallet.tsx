import { useCallback } from 'react';

import { useLogionChain } from '../logion-chain';

import Dashboard from '../common/Dashboard';

import UserRouter, {
    HOME_PATH,
    TRUST_PROTECTION_PATH,
    SETTINGS_PATH,
    RECOVERY_PATH,
    WALLET_PATH,
    locRequestsPath,
    VAULT_PATH,
    VTP_PATH,
} from "./UserRouter";
import { useUserContext } from "./UserContext";
import { useCommonContext } from '../common/CommonContext';
import Loader from "../common/Loader";

export default function ContextualizedWallet() {
    const { selectAddress, accounts, api } = useLogionChain();
    const { colorTheme, refresh } = useCommonContext();
    const { refreshRequests, vaultState, locsState } = useUserContext();

    const refreshAll = useCallback(() => {
        refresh(false);
        refreshRequests!(false);
    }, [ refresh, refreshRequests ]);

    if(selectAddress === null || accounts === null) {
        return null;
    }

    if(!locsState || locsState.discarded) {
        return <Loader />;
    }

    const userContext = api !== null ? <UserRouter /> : null;

    let menuTop = [
        {
            id: "home",
            text: "Home",
            to: HOME_PATH,
            exact: true,
            icon: {
                icon: {
                    id: 'home'
                },
                background: colorTheme.topMenuItems.iconGradient,
            },
            onClick: refreshAll,
        },
        {
            id: "wallet",
            text: "Wallet",
            to: WALLET_PATH,
            exact: false,
            icon: {
                icon: {
                    id: 'wallet'
                },
                background: colorTheme.topMenuItems.iconGradient,
            },
            onClick: refreshAll,
        },
        {
            id: "vault",
            text: "Vault",
            to: VAULT_PATH,
            exact: false,
            icon: {
                icon: {
                    id: 'vault'
                },
                background: colorTheme.topMenuItems.iconGradient,
            },
            onClick: refreshAll,
            disabled: !vaultState,
        },
        {
            id: "loc-collection",
            text: "Collections",
            to: locRequestsPath('Collection'),
            exact: false,
            icon: {
                icon: {
                    id: 'collection'
                },
                background: colorTheme.topMenuItems.iconGradient,
            },
            onClick: refreshAll,
        },
        {
            id: "loc-transaction",
            text: "Transactions",
            to: locRequestsPath('Transaction'),
            exact: false,
            icon: {
                icon: {
                    id: 'loc'
                },
                background: colorTheme.topMenuItems.iconGradient,
            },
            onClick: refreshAll,
        },
        {
            id: "loc-identity",
            text: "Identity",
            to: locRequestsPath('Identity'),
            exact: false,
            icon: {
                icon: {
                    id: 'identity'
                },
                background: colorTheme.topMenuItems.iconGradient,
            },
            onClick: refreshAll,
        },
    ];

    if (locsState?.isVerifiedThirdParty) {
        menuTop.push({
                id: "vtp",
                text: "Third Party LOC",
                to: VTP_PATH,
                exact: false,
                icon: {
                    icon: {
                        id: 'vtp-icon'
                    },
                    background: colorTheme.topMenuItems.iconGradient,
                },
                onClick: refreshAll,
            }
        )
    }
    return (
        <Dashboard
            menuTop={ menuTop }
            menuMiddle={[
                {
                    id: "protection",
                    text: "My Logion Protection",
                    to: TRUST_PROTECTION_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'shield',
                            hasVariants: true,
                        },
                        height: 'auto',
                        width: 'auto',
                    },
                    onClick: refreshAll,
                }
            ]}
            menuBottom={[
                {
                    id: "settings",
                    text: "Settings",
                    to: SETTINGS_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'settings'
                        },
                        background: colorTheme.bottomMenuItems.iconGradient,
                    },
                },
                {
                    id: "recovery",
                    text: "Recovery",
                    to: RECOVERY_PATH,
                    exact: true,
                    icon: {
                        icon: {
                            id: 'recovery'
                        },
                        background: colorTheme.recoveryItems.iconGradient,
                    },
                    onClick: refreshAll,
                }
            ]}
        >
            {userContext}
        </Dashboard>
    );
}
