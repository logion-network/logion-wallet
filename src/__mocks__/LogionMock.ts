import { AccountType, ValidAccountId, LogionNodeApiClass, LegalOfficerCase, Hash } from "@logion/node-api";
import { H256 } from "@logion/node-api/dist/types/interfaces";
import { It, Mock } from "moq.ts";
import type { Text, Compact, u128 } from '@polkadot/types-codec';
import type { SubmittableExtrinsic } from '@polkadot/api/promise/types';
import { TEST_WALLET_USER } from "src/wallet-user/TestData";

export const DEFAULT_LEGAL_OFFICER = mockValidPolkadotAccountId("5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"); // Alice
export const ANOTHER_LEGAL_OFFICER = mockValidPolkadotAccountId("5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"); // Bob
export const A_THIRD_LEGAL_OFFICER = mockValidPolkadotAccountId("5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y"); // Charlie

export function mockValidPolkadotAccountId(address: string): ValidAccountId {
    return mockValidAccountId(address, "Polkadot");
}

export function mockValidAccountId(address: string, type: AccountType): ValidAccountId {
    return {
        address,
        type,
        toKey: () => `${type}:${address}`,
    } as ValidAccountId;
}

export const api = new Mock<LogionNodeApiClass>();

api.setup(instance => instance.queries.getValidAccountId).returns((address: string, _type: string) => mockValidPolkadotAccountId(address));

const localPeerId = new Mock<Text>();
localPeerId.setup(instance => instance.toString()).returns("Mock peer ID");
api.setup(instance => instance.polkadot.rpc.system.localPeerId()).returnsAsync(localPeerId.object());

api.setup(instance => instance.polkadot.createType(It.IsAny())).returnsAsync({});

const locId = new Mock<Compact<u128>>();
api.setup(instance => instance.adapters.toLocId(It.IsAny())).returns(locId.object());

const hash = new Mock<H256>();
api.setup(instance => instance.adapters.toH256(It.IsAny())).returns(hash.object());

export function setupApiMock(setupFunction: (api: Mock<LogionNodeApiClass>) => void) {
    setupFunction(api);
}

export function mockSubmittable(): Mock<SubmittableExtrinsic> {
    return new Mock<SubmittableExtrinsic>();
}

export const CLOSED_IDENTITY_LOC_ID = "85833363768713528858922097642089825569";

export const CLOSED_IDENTITY_LOC: LegalOfficerCase = {
    owner: DEFAULT_LEGAL_OFFICER.address,
    requesterAddress: mockValidPolkadotAccountId("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    locType: 'Identity',
    closed: true,
    files: [
        {
            hash: Hash.fromHex("0xf35e4bcbc1b0ce85af90914e04350cce472a2f01f00c0f7f8bc5c7ba04da2bf2"),
            nature: Hash.fromHex("0x636af5586476273b2b9d9fc318e0d1f124f4e9591b7218da4e4e27f6b071cc94"),
            submitter: TEST_WALLET_USER,
            size: 42n,
            acknowledged: true,
        }
    ],
    metadata: [],
    links: [],
    collectionCanUpload: false,
}

export const OPEN_IDENTITY_LOC_ID = "195914524858768213081425411950368569411";

export const OPEN_IDENTITY_LOC: LegalOfficerCase = {
    owner: DEFAULT_LEGAL_OFFICER.address,
    requesterAddress: mockValidPolkadotAccountId("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    locType: 'Identity',
    closed: false,
    files: [],
    metadata: [],
    links: [],
    collectionCanUpload: false,
}

export const CLOSED_COLLECTION_LOC_ID = "195914524858768213081425411950368569411";

export const CLOSED_COLLECTION_LOC: LegalOfficerCase = {
    owner: DEFAULT_LEGAL_OFFICER.address,
    requesterAddress: mockValidPolkadotAccountId("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    locType: 'Collection',
    closed: true,
    files: [],
    metadata: [],
    links: [],
    collectionCanUpload: false,
}
