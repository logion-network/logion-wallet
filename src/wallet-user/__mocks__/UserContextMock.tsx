import { LogionNodeApiClass, Fees } from '@logion/node-api';

import { twoLegalOfficers } from "../../common/TestData";
import { TEST_WALLET_USER } from '../TestData';
import { AccountTokens, DirectoryClient, NoProtection, ProtectionState, LegalOfficerClass, DraftRequest } from '@logion/client';
import { AxiosFactory } from '@logion/client/dist/AxiosFactory.js';
import { ComponentFactory } from '@logion/client/dist/ComponentFactory.js';
import { LegalOfficerEndpoint, LogionClientConfig, SharedState } from '@logion/client/dist/SharedClient.js';
import { NetworkState } from '@logion/client/dist/NetworkState.js';
import { UserContext, ProtectionRequestParams } from '../UserContext';
import { BalanceState } from '@logion/client/dist/Balance.js';
import {
    LocsState,
    PendingRequest,
    ClosedLoc,
    RejectedRequest,
    ClosedCollectionLoc,
    OpenLoc, VoidedCollectionLoc, VoidedLoc
} from "@logion/client";
import { LocType, ValidAccountId } from "@logion/node-api";
import { Mock } from 'moq.ts';

export let createTokenRequest = () => null;

export let refreshRequests = jest.fn();

export let createProtectionRequest = jest.fn().mockReturnValue(Promise.resolve());

export function estimateFeesCreateProtectionRequest(_: ProtectionRequestParams): Promise<Fees> {
    return Promise.resolve(Fees.zero());
}

export let activateProtection = jest.fn().mockReturnValue(Promise.resolve());

export let claimRecovery = jest.fn().mockReturnValue(Promise.resolve());

export let api = new Mock<LogionNodeApiClass>();

export const DEFAULT_SHARED_STATE: SharedState = {
    axiosFactory: {} as AxiosFactory,
    componentFactory: {} as ComponentFactory,
    config: {

    } as LogionClientConfig,
    directoryClient: {} as DirectoryClient,
    legalOfficers: twoLegalOfficers,
    allLegalOfficers: twoLegalOfficers,
    networkState: {} as NetworkState<LegalOfficerEndpoint>,
    nodeApi: api.object(),
    tokens: {} as AccountTokens,
    currentAccount: {
        address: TEST_WALLET_USER,
        type: "Polkadot"
    } as unknown as ValidAccountId,
}

export let protectionState: ProtectionState | undefined = {} as NoProtection;

export function setProtectionState(state: ProtectionState | undefined) {
    protectionState = state;
}

export let recoveredBalanceState: BalanceState | undefined = {} as BalanceState;

export function setRecoveredBalanceState(state: BalanceState | undefined) {
    recoveredBalanceState = state;
}

export let mutateRecoveredBalanceState = jest.fn().mockReturnValue(Promise.resolve());

export let locsState: Partial<LocsState> | undefined = undefined;

export function setOpenedLocRequests(requests: any[]) {
    locsState = {
        get openLocs(): Record<LocType, OpenLoc[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setClosedLocRequests(requests: any[]) {
    locsState = {
        get closedLocs(): Record<LocType, (ClosedLoc | ClosedCollectionLoc)[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setVoidedLocs(requests: any[]) {
    locsState = {
        get voidedLocs(): Record<LocType, (VoidedLoc | VoidedCollectionLoc)[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setPendingLocRequests(requests: any[]) {
    locsState = {
        get pendingRequests(): Record<LocType, PendingRequest[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setRejectedLocRequests(requests: any[]) {
    locsState = {
        get rejectedRequests(): Record<LocType, RejectedRequest[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setDraftLocRequests(requests: any[]) {
    locsState = {
        get draftRequests(): Record<LocType, DraftRequest[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setOpenVerifiedIssuerLocs(transactions: any[], collections: any[]) {
    locsState = {
        get openVerifiedIssuerLocs(): Record<LocType, OpenLoc[]> {
            return {
                "Transaction": transactions,
                "Collection": collections,
                "Identity": [],
            }
        }
    }
}
export function setClosedVerifiedIssuerLocs(transactions: any[], collections: any[]) {
    locsState = {
        get closedVerifiedIssuerLocs(): Record<LocType, ClosedLoc[]> {
            return {
                "Transaction": transactions,
                "Collection": collections,
                "Identity": [],
            }
        }
    }
}

export function setHasNonVoidIdentityLoc(legalOfficers: LegalOfficerClass[]) {
    locsState = {
        legalOfficersWithNonVoidIdentityLoc: legalOfficers
    }
}

export function setLocsState(locsStateMock: LocsState) {
    locsState = locsStateMock;
}

export let mutateLocsState = jest.fn().mockReturnValue(Promise.resolve());

export function useUserContext() {
    return {
        refreshRequests,
        createProtectionRequest,
        estimateFeesCreateProtectionRequest,
        activateProtection,
        claimRecovery,
        protectionState,
        recoveredBalanceState,
        mutateRecoveredBalanceState,
        locsState,
        mutateLocsState,
    } as unknown as Partial<UserContext>;
}

export function setCreateTokenRequest(callback: any) {
    createTokenRequest = callback;
}

export function setRefreshRequests(func: any) {
    refreshRequests = func;
}

export function setCreateProtectionRequest(callback: any) {
    createProtectionRequest = callback;
}

export function setMutateLocsState(callback: any) {
    mutateLocsState = callback;
}

export function UserContextProvider() {
    return (
        <div/>
    );
}
