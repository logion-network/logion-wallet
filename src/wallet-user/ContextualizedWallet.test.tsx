import { setLocsState } from "./__mocks__/UserContextMock";

jest.mock('../logion-chain');
jest.mock('../common/CommonContext');
jest.mock('./UserContext');

import ContextualizedWallet from './ContextualizedWallet';
import { apiMock, setContextMock } from '../logion-chain/__mocks__/LogionChainMock';
import { shallowRender } from '../tests';
import { LocsState } from "@logion/client";
import { mockValidPolkadotAccountId } from 'src/__mocks__/LogionMock';

describe("ContextualizedWallet", () => {

    setLocsState({} as LocsState)

    const accountId = mockValidPolkadotAccountId("address");

    test('Given disconnected and accounts then empty', () => {
        setContextMock({
            api: null,
            accounts: {
                current: {
                    accountId
                }
            }
        });
    
        const tree = shallowRender(<ContextualizedWallet />);
    
        expect(tree).toMatchSnapshot();
    });
    
    test('Given connected and account, then renders with router', () => {
        setContextMock({
            api: apiMock,
            accounts: {
                current: {
                    accountId
                }
            }
        });

        const tree = shallowRender(<ContextualizedWallet />);
    
        expect(tree).toMatchSnapshot();
    });

    test('Given connected and no account then renders without router', () => {
        setContextMock({
            api: apiMock,
            accounts: null
        });
    
        const tree = shallowRender(<ContextualizedWallet />);
    
        expect(tree).toMatchSnapshot();
    }); 
});
