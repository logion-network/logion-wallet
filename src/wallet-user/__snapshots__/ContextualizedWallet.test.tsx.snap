// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextualizedWallet Given connected and no metadata, when rendering, then show account, no connect button, no node info, tokenization 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "onClick": [Function],
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "onClick": [Function],
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "home",
          },
        },
        "id": "home",
        "onClick": [Function],
        "text": "Home",
        "to": "/user",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "onClick": [Function],
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "disabled": true,
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "vault",
          },
        },
        "id": "vault",
        "onClick": [Function],
        "text": "Vault",
        "to": "/user/vault",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "collection",
          },
        },
        "id": "loc-collection",
        "onClick": [Function],
        "text": "Collections",
        "to": "/user/loc/collection",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "loc",
          },
        },
        "id": "loc-transaction",
        "onClick": [Function],
        "text": "Transactions",
        "to": "/user/loc/transaction",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "identity",
          },
        },
        "id": "loc-identity",
        "onClick": [Function],
        "text": "Identity",
        "to": "/user/loc/identity",
      },
    ]
  }
>
  <UserRouter />
</Dashboard>
`;

exports[`ContextualizedWallet Given connected, when rendering, then show account, no connect button and tokenization 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "onClick": [Function],
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "onClick": [Function],
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "home",
          },
        },
        "id": "home",
        "onClick": [Function],
        "text": "Home",
        "to": "/user",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "onClick": [Function],
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "disabled": true,
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "vault",
          },
        },
        "id": "vault",
        "onClick": [Function],
        "text": "Vault",
        "to": "/user/vault",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "collection",
          },
        },
        "id": "loc-collection",
        "onClick": [Function],
        "text": "Collections",
        "to": "/user/loc/collection",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "loc",
          },
        },
        "id": "loc-transaction",
        "onClick": [Function],
        "text": "Transactions",
        "to": "/user/loc/transaction",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "identity",
          },
        },
        "id": "loc-identity",
        "onClick": [Function],
        "text": "Identity",
        "to": "/user/loc/identity",
      },
    ]
  }
>
  <UserRouter />
</Dashboard>
`;

exports[`ContextualizedWallet Given connecting, when rendering, then show account, no button, no node info 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "onClick": [Function],
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "onClick": [Function],
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "home",
          },
        },
        "id": "home",
        "onClick": [Function],
        "text": "Home",
        "to": "/user",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "onClick": [Function],
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "disabled": true,
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "vault",
          },
        },
        "id": "vault",
        "onClick": [Function],
        "text": "Vault",
        "to": "/user/vault",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "collection",
          },
        },
        "id": "loc-collection",
        "onClick": [Function],
        "text": "Collections",
        "to": "/user/loc/collection",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "loc",
          },
        },
        "id": "loc-transaction",
        "onClick": [Function],
        "text": "Transactions",
        "to": "/user/loc/transaction",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "identity",
          },
        },
        "id": "loc-identity",
        "onClick": [Function],
        "text": "Identity",
        "to": "/user/loc/identity",
      },
    ]
  }
>
  <UserRouter />
</Dashboard>
`;

exports[`ContextualizedWallet Given connection error, when rendering, then show account, no button, no node info 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "onClick": [Function],
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "onClick": [Function],
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "home",
          },
        },
        "id": "home",
        "onClick": [Function],
        "text": "Home",
        "to": "/user",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "onClick": [Function],
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "disabled": true,
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "vault",
          },
        },
        "id": "vault",
        "onClick": [Function],
        "text": "Vault",
        "to": "/user/vault",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "collection",
          },
        },
        "id": "loc-collection",
        "onClick": [Function],
        "text": "Collections",
        "to": "/user/loc/collection",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "loc",
          },
        },
        "id": "loc-transaction",
        "onClick": [Function],
        "text": "Transactions",
        "to": "/user/loc/transaction",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "identity",
          },
        },
        "id": "loc-identity",
        "onClick": [Function],
        "text": "Identity",
        "to": "/user/loc/identity",
      },
    ]
  }
>
  <UserRouter />
</Dashboard>
`;

exports[`ContextualizedWallet Given disconnected, when rendering, then show disconnected and connect button 1`] = `null`;

exports[`ContextualizedWallet Given no injected account, when rendering, then nothing displayed 1`] = `
<Dashboard
  menuBottom={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "settings",
          },
        },
        "id": "settings",
        "text": "Settings",
        "to": "/user/settings",
      },
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#7a90cb",
            "to": "#3b6cf4",
          },
          "icon": Object {
            "id": "recovery",
          },
        },
        "id": "recovery",
        "onClick": [Function],
        "text": "Recovery",
        "to": "/user/recovery",
      },
    ]
  }
  menuMiddle={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "height": "auto",
          "icon": Object {
            "hasVariants": true,
            "id": "shield",
          },
          "width": "auto",
        },
        "id": "protection",
        "onClick": [Function],
        "text": "My Logion Protection",
        "to": "/user/protection",
      },
    ]
  }
  menuTop={
    Array [
      Object {
        "exact": true,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "home",
          },
        },
        "id": "home",
        "onClick": [Function],
        "text": "Home",
        "to": "/user",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "wallet",
          },
        },
        "id": "wallet",
        "onClick": [Function],
        "text": "Wallet",
        "to": "/user/wallet",
      },
      Object {
        "disabled": true,
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "vault",
          },
        },
        "id": "vault",
        "onClick": [Function],
        "text": "Vault",
        "to": "/user/vault",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "collection",
          },
        },
        "id": "loc-collection",
        "onClick": [Function],
        "text": "Collections",
        "to": "/user/loc/collection",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "loc",
          },
        },
        "id": "loc-transaction",
        "onClick": [Function],
        "text": "Transactions",
        "to": "/user/loc/transaction",
      },
      Object {
        "exact": false,
        "icon": Object {
          "background": Object {
            "from": "#3b6cf4",
            "to": "#6050dc",
          },
          "icon": Object {
            "id": "identity",
          },
        },
        "id": "loc-identity",
        "onClick": [Function],
        "text": "Identity",
        "to": "/user/loc/identity",
      },
    ]
  }
>
  <UserRouter />
</Dashboard>
`;
